index.rst

```diff
@@ -2,24 +2,206 @@
+This also means we are constantly receiving feedback from users and making adjustments to the software based on those feedback.
+
+In terms of requirements verification, the fact that software is constantly changing means we need to be constantly verifying requirements.
+In order to achieve a constant stream of new features and implement a user-feedback mechanism, we rely on Continuous Integration and Continuous Delivery (CI/CD).
+
+The CI/CD discipline originates from the `Agile methodology <https://www.agilealliance.org/agile101/>`__ as a way to respond to the rapid delivery of new software on a project.
+Overall, the idea is that each new software feature (or bug-fixes) must also contains a set of tests that verify the functionality.
+In general, tests are separated in two different categories; unit tests and integration tests.
+
+Unit tests are small scale tests designed to verify a particular functionality of a software component.
+They usually run in isolation utilizing mocks for other components when required.
+These are the lowest level of testing and even though they usually exercise the requirements, they are not suitable for verifying requirements, either because they execute in isolated environments, because generating reports are not straightforward or else.
+
+Integration tests usually consist of standard or custom operations that exercise a particular component or a group of components in a systematic way.
+These tests are designed to run both on testing and production environments.
+Because these tests are executed in an environment hosting the integrated system, they provide the perfect conditions to generate requirements verification artifacts.
+
+A CI/CD system is an integral part of Telescope and Site Software deliverables, alongside a suite of integration tests, and a reporting system that are designed to verify the system integrity after every new release and for daily checkouts prior to night-time observations.
+
+We plan to use this suite of integration tests to verify the majority of Telescope and Site Software requirements, alongside any affiliated requirements that can be verified in the process and use the reporting system to generate verification artifacts.
+In the following sections we will go over the details of the different steps in the testing process and reporting products.
+
+Testing Suite
+=============
+
+In this section, we discuss the collection of tests that are most relevant for the formal requirement verification.
+.. Still, as we mentioned earlier, there are a whole selection of unit tests that are used early on in the development process to safeguard the integrity of each software components.
+It should be remembered that there are minimal functionality tests and other categories of tests completed at various stages of the system lifecycle to safeguard the integrity of each component in addition to the tests discussed in this section.
```

Comment:
Consider:
One should be reminded that in addition to the tests discussed in this section,  there are also minimal functionality tests and other categories of tests completed at various stages of the system lifecycle to safeguard the integrity of each component in the system.